core/src/software/aws/toolkits/core/utils/Waiter.kt

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.core.utils
+
+import software.amazon.awssdk.awscore.exception.AwsServiceException
+import java.time.Duration
+
+fun <T> wait(
```

Comment:
Should prolly comment each of these params since this is quite a big function signature

Reply:
Done.